cli/api/dbadapters/execution_sql.ts

```diff
@@ -196,17 +196,29 @@ from (${query}) as insertions`;
       }
     }
 
-    return `create or replace ${table.materialized ? "materialized " : ""}${this.tableTypeAsSql(
-      this.baseTableType(table.enumType)
-    )} ${this.resolveTarget(table.target)} ${
-      table.bigquery && table.bigquery.partitionBy
-        ? `partition by ${table.bigquery.partitionBy} `
-        : ""
-    }${
-      table.bigquery && table.bigquery.clusterBy && table.bigquery.clusterBy.length > 0
-        ? `cluster by ${table.bigquery.clusterBy.join(", ")} `
-        : ""
-    }${options.length > 0 ? `OPTIONS(${options.join(",")})` : ""}as ${table.query}`;
+    let statement = "create ";
+    if (table.enumType !== dataform.TableType.INCREMENTAL) {
+      statement += "or replace ";
```

Comment:
From the bug - I think we still want to create or replace in the case that it is incremental AND it's not projected AND it's full refresh?

Reply:
We should also add a test for this case

Reply:
What is projected? Have added the full refresh.

This is tested, but not obviously, across https://github.com/dataform-co/dataform/blob/c55ca6bcc79b1087545655605c7855447d091a45/tests/api/api.spec.ts#L463

Reply:
*protected!

Reply:
Ah right! Done.

I had to update the protected logic, such as removing the error that we test for here https://github.com/dataform-co/dataform/blob/c55ca6bcc79b1087545655605c7855447d091a45/tests/api/api.spec.ts#L109. But not it should act as the incremental spec suggests.

Reply:
Right, sorry I think this has got really confusing, a few thoughts.

1. leave this method as `createOrReplace` and don't pass in run options or do all that logic.
2. Add a `createIfNotExists` method next to it (they can share some private impl if you want, but I wouldn't bother)
3. Update the logic in publishTasks, only for the incremental tables, to do what you want to do and call `createIfNotExists` by default, and `createOrReplace` only when `!protected && fullRefresh`

The original method was clear what it did, but now you've moved this logic into it, I'd say it's much harder to follow what's going on in `publishTasks`.